src/tickets/types.ts

```diff
@@ -0,0 +1,138 @@
+import {FirebaseFirestoreTypes} from '@react-native-firebase/firestore';
+
+enum TravelRightStatus {
+  UNSPECIFIED = 0,
+  RESERVED = 1,
+  ORDERED = 2,
+  PAID_FOR = 3,
+  UNUSED = 4,
+  ACTIVATED = 5,
+  PARTIALLY_USED = 6,
+  USED = 7,
+  ARCHIVED = 8,
+  OTHER = 9,
+  CANCELLED = 20,
+  REFUNDED = 21,
+}
+
+export type TravelRight = {
+  id: string;
+  status: TravelRightStatus;
+  type:
+    | 'PreActivatedSingleTicket'
```

Comment:
Hmm, these types, are they actually necessary to store in Firestore? 

I'm thinking especially of the distinction between single ticket, period ticket and unknown ticket. Whether the ticket is activated or not must be in Firestore.

My thinking is that other product information is not included on the fare contract, but fetched from reference data by using the `fareProductRef`. And information whether it is a single ticket or period ticket product will probably also be present there (either directly or as metadata). Also if the `fareProductRef` does not point to a product we have in our reference data it will be an unknown ticket. My thinking is just that we should have a consistent pattern for what the source of information is, but of course there may be reasons making it necessary to store this in Firestore which I am unaware of now.

Reply:
The generic nature of the travel rights in the id-based backend. As of now, you're right we could probably deduce from the data if it is a single ticket or period ticket. But later when we get things like travel cards, and "klippekort", not to mention other modes of transport or travel rights from other providers (than AtB), the deducing and complexity becomes intractable. So our thinking long-term is that if we let the backend deduce the ticket type, it allows us to make assumption about which data is included in the payload depending on the type.

Reply:
Sounds good, as long as it is a deliberate decision I'm happy!